martypy/ClientMV2.py

```diff
@@ -400,6 +401,81 @@ def preException(self, isFatal: bool) -> None:
+                if add_ons['whoAmITypeCode'] in disco:
+                    return True
+                else:
+                    raise MartyCommandException("The add on name passed in is not a valid disco add on.")
+        raise MartyCommandException("The add on name passed in is not a valid add on.")
+
+    def valid_hex(self, hexc: str) -> bool :
+        regex = "^([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$"
+        p = re.compile(regex)
+        if not hexc:
+            return False
+        if (re.search(p,hexc)):
+            return True
+        return False
+
+    def disco_off(self, add_on: str = 'all') -> bool :
+        pattern = '01'
+        if self.valid_addon(add_on):
+            return self.ricIF.cmdRICRESTRslt(f"elem/{add_on}/json?cmd=raw&hexWr={pattern}")
+
+    def disco_pattern(self,  pattern: str, add_on: str ) -> bool :
+        if self.valid_addon(add_on):
+            return self.ricIF.cmdRICRESTRslt(f"elem/{add_on}/json?cmd=raw&hexWr={pattern}")
+
+    def disco_cmd_hex(self, hexc: str, add_on: str, region: int) -> bool:
+        if region == 'all':
+            region = '02'
+        else:
+            region = '040' + str(region)
+        if self.valid_addon(add_on):
+            return self.ricIF.cmdRICRESTRslt(f"elem/{add_on}/json?cmd=raw&hexWr={region}{hexc}")
+
+    def disco_color(self, color: Union[int,str,tuple], add_on: str, region: Union[int,str] = 'all') -> bool:#mayb switch  color and add_on
+        default_colors = {
+            'white':'FFFFFF',
+            'red':'FF0000',
+            'blue':'0000FF',
+            'yellow':'FFFF00',
+            'green':'008000',
+            'teal':'008080',
+            'pink':'800080',
+            'purple':'060014',
+            'orange':'0f0200'
+        }
+        if type(color) is str:
+            try:
+                hex_color = default_colors[color.lower()]
+                return self.disco_cmd_hex(hex_color, add_on,region)
+            except KeyError:
+                if color[0] == '#':
+                    color = color[1:]
+                if self.valid_hex(color): # add other params of hex code here
+                    return self.disco_cmd_hex(color, add_on,region)
+                else:
+                    raise MartyCommandException("Color specified is not a valid hex code or default color")
+        if type(color) is tuple:
+            if len(color)!= 3:
+                raise MartyCommandException("RGB tuple must be 3 numbers, please enter valid color.")
+            hex_color = '%02x%02x%02x' % color
+            return self.disco_cmd_hex(hex_color, add_on,region)
+
+    def disco_group(self, function: str, group: set = {"00000087","00000088","00000089"}, params: dict = {}) :
+        result = []
+        for add_ons in self.get_add_ons_status().values():
+            if add_ons['whoAmITypeCode'] in group:
+                addon_name = add_ons['name']
+                result.append(function(**params,**{'add_on':addon_name}))
```

Comment:
```suggestion
                result.append(function(add_on=addon_name, **params))
```
I think this would also work, no? And it looks a bit nicer I think.